### ISSUE TITLE
Fix asan errors with strEQc/memEQc

### ISSUE THREAD

**Issue by rurban** [perl11/cperl#207]
Add a local conf and smoker, and maybe even a public test to travis.
probe for asan.

use macro variants for `strEQc(s,c), strNEc(s,c)` just with the safe slow strcmp, and without needing the size of the right constant, where we can guarantee that the left side is not at a page boundary. (which we can due to our arena).
See also GH #189 

Also several other minor asan issues:
- use-after-free in Dynaloader (ReadKey probe with -DDEBUG_LEAKING_SCALAR)
- heap-overflow in gv_fetchfile (t/comp/parser.t)
- signature: gmpb -Dkv -e'sub t151($a,$b=do{my $foo;1},$c=1){} t151($x,$x,$x);'
- heap-buffer-overflow in XSLoader
- invalid memEQc in toke.c
- B-C global-buffer-overflow with dynamic COW strings (27,28,43,44,45,49), wrong savepvn args

**Comment by rurban** [comment]
Remaining failing tests:

```
run/switchx.t                                                   (Wstat: 0 Tests: 2 Failed: 0)
  Parse errors: Tests out of sequence.  Found (7) but expected (1)
                Tests out of sequence.  Found (8) but expected (2)
                No plan found in TAP output
```

missing -l cmdline opt, parsed from a #! line (wrong strEQc in open_script(), replaced by memEQc). fixed with 4a6d30f717fdf4c26a80c6b8811da197d7fa5371

```
../lib/perl5db.t                                                (Wstat: 0 Tests: 133 Failed: 5)
  Failed tests:  19-20, 48, 119, 122
```

fixed with the XSLoader heap-overflow fix in 8e6279e67680656e70350f31b6eef3730f4051b2

signatures padintro heap-overflow `++items`: ck_entersub_args_signature op.c:12409

**Comment by rurban** [comment]
Merged with f45debbab1aada53dc3874864b647c7b74218201
